### ISSUE TITLE
Announcement: Finite Galaxy

### ISSUE THREAD

**Issue by finite-galaxy** [endless-sky/endless-sky#4082]
Greetings!

When I first tried out Endless Sky I was immediately hooked. The quality of both code and content is high, the resulting game addictive. I've been following the github version for about two years now, and have seen it slowly get even better. However, there are also things I dislike or would have done differently.

After toying with the idea for quite a while, I finally decided to start my own derivative on October 18. Reasons for forking include frustration about the limitations of plug-ins, the lessened pace of development this year, and disappointment that useful proposals such as #3502 or #3870 were ignored.
My version of Endless Sky is called **Finite Galaxy**.

Both games can be installed alongside and played independently of each other. Although derived from the same source code and basically using the same content, Finite Galaxy and Endless Sky have diverged and are no longer compatible. If you transpose a save game from one to the other, you are likely to encounter hundreds of errors.

In the past two months I've made hundreds of commits. A selection of changes:
* GUI tweaks to display more stats
* ship categories are based on the ship's `total mass = hull mass + outfit space + cargo space`
* jump fuel is based on ship's effective mass, ships have 0 fuel capacity by default, fuel pod size is reduced to 1
* introduced core space, reserved for energy generators, shields and hull systems, and hyperdrives (`outfit space = core space + engine space + weapon space`)
* installed outfits can no longer be plundered by default; outfits in cargo still can
* minimum depreciation value raised to 50%, time lowered to one year
* redistributed outfits and ships in human space to achieve more regional specialization
* removed news, portraits, and conversation scenes (the supporting code is still present, to allow them to be re-added in plug-ins)
* distances from planets to the sytem's centre are trebled; as a result space feels larger, thrusters are more desirable, and players won't always land immediately in the middle of a space fight
* non-missile weapons have their weapon range increased by about a third
* ships consume a small amount of energy based on the number of bunks (“life support”)
* banks are no longer present on space stations, pirate planets, or alien worlds
* numerous map, category, fleets, storyline, outfits, ship, and other tweaks

The purpose of Finite Galaxy is not to compete with Endless Sky or split its community; it is to offer my vision of what the game could be and to experiment with features that are probably too controversial to be included in the main distribution. Finite Galaxy is, of course, free and open source software under the same licence as Endless Sky; if there is something you like, you can include that in Endless Sky, or in a different branch, fork, or project.

To summarize, Finite Galaxy is a similar game with different details, a slightly more challenging version for people who already know Endless Sky; feel free to try it out!

You can find it at https://github.com/finite-galaxy/finite-galaxy

**Comment by finite-galaxy** [comment]
And because an image says more than a thousand words, here are some screenshots:

* shop panel:
![screenshot from 2018-12-17 13-17-03](https://user-images.githubusercontent.com/44258906/50162746-5ba4cb80-02df-11e9-9ef9-d0f4ce66c8fc.png)

* map at game start:
![screenshot from 2018-12-17 14-01-40](https://user-images.githubusercontent.com/44258906/50162747-5ba4cb80-02df-11e9-8c94-d80a1b17c656.png)

* map after completing storyline missions:
![screenshot from 2018-12-17 14-03-10](https://user-images.githubusercontent.com/44258906/50162748-5ba4cb80-02df-11e9-8451-80b3c15fd5c4.png)

* per shot and per second displayed on same line:
![screenshot from 2018-12-17 12-38-33](https://user-images.githubusercontent.com/44258906/50162744-5b0c3500-02df-11e9-8f51-a90757e8906c.png)

* ships display miscellaneous stats:
![screenshot from 2018-12-18 12-48-52](https://user-images.githubusercontent.com/44258906/50162750-5c3d6200-02df-11e9-81d1-5a96071c7894.png)

* ship shapes:
![ship_shapes](https://user-images.githubusercontent.com/44258906/50162997-cf46d880-02df-11e9-8a46-650ac77e9375.png)

* and more swizzle options:
![wardragon_swizzles](https://user-images.githubusercontent.com/44258906/50163027-d968d700-02df-11e9-8095-df401231a0fa.png)

**Comment by Amazinite** [comment]
Closing this because it's really just an advertisement for a TC mod.

**Comment by Zitchas** [comment]
Looks like a lot of content. Should probably be posted to a plug-in channel / section rather than issues though. Always nice to see more content being made for ES!

**Comment by TheMarksman-ES** [comment]
>To clarify, it's not a plug-in, it's a fork (i.e. independent game) with numerous code and data edits.

>Whilst I agree it's not really an issue, there is not a dedicated plug-in tab here.

I beg to differ, but GitHub here is used for game development, and not a free for all community sharing site.

Your best bet would be the mods section on Steam.

**Comment by finite-galaxy** [comment]
Well, I'm not active on Steam, Google groups, Discord, etc, nor do I intend to.
The reason I posted this announcement here on github, is because I believe having one or more forks to experiment with multiple new ideas could help with development of Endless Sky's main distribution.

**Comment by Pointedstick** [comment]
Looks like you put a lot of work into this!

However, I hope you fully understand the amount of work needed to maintain a fork of an active project. Endless Sky continues to be developed, and the longer time goes on, the more difficulty you will have in integrating the latest changes into your fork. Most forks suffer from this phenomenon of continuously increasing maintenance burden, and usually the development dramatically slows or stops within 4 years. I have seen this happen time and time again. It's almost an iron law of forks. The true tragedy is that when a fork dies, all the work that went into it dies too, whereas if only the work had been contributed to the original project, it would live on, immortal.

Therefore I strongly encourage you to work on submitting these changes as PRs. Many of them sound like great ideas to me, and I think that quite a few are likely to be non-controversial, such as consolidating the per-shot/per-second stats on one line and I imagine most of the small fixes.

I also really like the idea of life support energy, mass-dependent hyperspace jump fuel requirements, and increasing stellar object and warp-in distances from the system center.

Try upstream development, you'll like it! :)

**Comment by jostephd** [comment]
@finite-galaxy While I'm glad to see new ideas being worked on, and look forward to playtesting them, I would advise against mass-renaming [function names](https://github.com/finite-galaxy/finite-galaxy/commit/313eb47125c8fc67a86ffe76258cdf1f36e1043c), [files](https://github.com/finite-galaxy/finite-galaxy/commit/f94e29ae878dcf4525f918aef7ec1fc8c42bf7cc), and [directories](https://github.com/finite-galaxy/finite-galaxy/commit/4038e14c6c2889bea95b0cf217a0e6da65d785b9). By doing all these trivial changes you're effectively forcing a split: few patches will ever migrate between endless-sky and finite-galaxy because the conflicts will be too onerous to resolve. I would strongly recommend to keep MZ's directory structure and symbol names to allow patches to migrate back and forth between endless-sky and finite-galaxy; that's the best way for _both_ projects to thrive.

Great choice of name, by the way :grin:

**Comment by DingusShingleton** [comment]
>  life support energy, mass-dependent hyperspace jump fuel requirements, and increasing stellar object and warp-in distances from the system center

These all sound like great mechanics for an optional, more hardcore mode of gameplay that increases the realism a bit and makes you have to really think about your options a bit more. For example, 100 credits/bunk is certainly a cost, but if you are trying to board as a means of income, there is never a time when paying the extra 100 is not an automatic yes.  If each bunk costs energy, all of a sudden you really need to think about the cost of adding bodies (though I think an empty bunk shouldn't cost anything).

**Comment by finite-galaxy** [comment]
The split effectively occurred two months ago, at the moment I started Finite Galaxy. Basically I took all the files and went my own way from there. Finite Galaxy is not a branch of Endless Sky, it's practically a separate project; derived from the same source, yes, but diverging.  I'm not interested in maintaining compatibility or in following Endless Sky changes. 
If I see something I like, I'll manually incorporate it, if not, then not. And if there is something you like in Finite Galaxy, feel free to take it. 

Finite Galaxy is not just Endless Sky with some changes. The vision is different, as are some fundamental principles. E.g. @endless-sky spent a lot of time to preserve backward compatibility with earlier versions, which is great from a player perspective. It is something I value and can appreciate, but also something I intentionally axed, because it is a burden to experimentation.

> I would advise against mass-renaming function names, files, and directories.

If I rename something, it's because it makes it for me personally more logical or easier to work. Both `sounds/` and `source/` start with the same letter, which is inconvenient; now I've renamed the latter to `code/`, I can simply type in my terminal `c` and hit Tab; it might sound trivial, but when you have to type it hundreds of times, it really helps.
Likewise, the `ships.txt` and `variant.txt` were too large for me to work effectively, so I reordered all ships and variants into separate data files per shipyard or species. 

And yes, I occassionally rename content as well, if I think something a more appropiate name. E.g. I disliked the Rainmaker, Gunboat, Frigate, Cruiser, and Carrier names, so I renamed those to, respectively, Mercury, Mars, Venus, Saturn, and Jupiter. And "New Boston" is now called "Endless Sky" (a suitable name for a starting location, don't you think?).

> Try upstream development, you'll like it! :)

Thanks but no thanks. As I wrote in my initial announcement, I've been following the Endless Sky github for about two years now. Proposing and discussing pull requests takes a lot more time than just making the changes. Some pull requests have been open for more than a year. Besides, I don't necessarily agree with all the changes made, nor do I expect others to accept all of my ideas.
If I want to change something, I can do it immediately, without having to wait months to get something reviewed and spent hours discussing a proposal to convince others. Even if I am the only one who'll play Finite Galaxy, I still consider it a better use of my time than trying to contribute here; no offence. 

> I also really like the idea of life support energy, mass-dependent hyperspace jump fuel requirements, and increasing stellar object and warp-in distances from the system center.

The code changes themselves are quite straightforward, but it also requires you to adjust every single ship/ringworld. 

> These all sound like great mechanics for an optional, more hardcore mode of gameplay that increases the realism a bit and forces you to really consider your choices.

Yes, bigger does not necessarily mean better in Finite Galaxy. (Also, extra bunks requires cargo space instead of outfit space.)
Another idea I might try out is making fuel cost credits (e.g. 1 credit per fuel in spaceports, 2 credits when hailing a ship in space). And something else I'm contemplating is limiting the commodities for sale on planets; e.g. on a farming planet you can only buy food.

> Great choice of name, by the way grin

It makes it immediately clear the game is similar yet different.

**Comment by Pointedstick** [comment]
I wish you good luck, then. Where practically everyone else has failed in their fork of actively developed software, may you succeed! It would be a real shame if you burn out in a few years and all your hard work is lost forever. :(

**Comment by finite-galaxy** [comment]
Pug. They are quite even-handed, so when they send two Arfectas and open the Eye for the Wanderers, they simultaneously help the Unfettered by linking "Ehma Ti" "Prakacha'a".
Likewise, the Pug open a wormhole between Minkar and Procyon the moment they flee human space.

**Comment by finite-galaxy** [comment]
Anyway, I wish everyone a Merry Christmas!
